iModelCore/ECDb/ECDb/ECSql/ECSqlPreparer.cpp

```diff
@@ -513,14 +513,15 @@ void ECSqlExpPreparer::RemovePropertyRefs(ECSqlPrepareContext& ctx, ClassRefExp
     ctx.GetSelectionOptionsR().Clear();
     for (auto& parentPropertyExp : parentPropertyExps)
         {
-        PropertyNameExp const* propertyName = parentPropertyExp->GetAsCP<PropertyNameExp>();
-        if (propertyName->IsPropertyRef())
+        PropertyNameExp const* propertyNameExp = parentPropertyExp->GetAsCP<PropertyNameExp>();
+        if (propertyNameExp->IsPropertyRef())
             continue;
-        if (propertyName->IsVirtualProperty())
+        if (propertyNameExp->IsVirtualProperty())
             break;
 
-        if (&classMap == &propertyName->GetPropertyMap().GetClassMap())
-            ctx.GetSelectionOptionsR().AddProperty(propertyName->GetPropertyMap());
+        const RangeClassRefExp* classRefExp = propertyNameExp->GetClassRefExp();

```

Comment:
Can you quickly explain this piece? So we did not have to adjust that selected properties dictionary after all?

Reply:
In `SELECT a.Prop FROM test.Foo a , test.Foo b ` we only like to select `a.Prop` but before this change` b.Prop` get selected as well as Prop is in both classes, and it was included in select column list. Instead of comparing alias qualified name a shortcut is to compare `a.Prop` resolved class ref which would be `test.Foo a` and so we just compare pointers.

Reply:
Understood, sounds good, thanks!